### PR TITLE
[Merged by Bors] - feat(FieldTheory/Adjoin): add `exists_finset_of_mem_adjoin`

### PR DESCRIPTION
**Pull request by acmepjz**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by erdOne** [comment]
maintainer merge

File: Mathlib/FieldTheory/Adjoin.lean [hunk 1]
```diff
@@ -799,6 +799,15 @@ theorem exists_finset_of_mem_supr'' {ι : Type*} {f : ι → IntermediateField F
...
 #align intermediate_field.exists_finset_of_mem_supr'' IntermediateField.exists_finset_of_mem_supr''
 
+theorem exists_finset_of_mem_adjoin {S : Set E} {x : E} (hx : x ∈ adjoin F S) :
+    ∃ T : Finset E, (T : Set E) ⊆ S ∧ x ∈ adjoin F (T : Set E) := by
+  simp_rw [← biSup_adjoin_simple S, ← iSup_subtype''] at hx
+  obtain ⟨s, hx'⟩ := exists_finset_of_mem_iSup hx
+  refine ⟨s.map ⟨Subtype.val, Subtype.val_injective⟩, by simp, SetLike.le_def.mp ?_ hx'⟩
+  simp_rw [Finset.coe_map, Function.Embedding.coeFn_mk, iSup_le_iff, adjoin_le_iff,
+    Set.le_eq_subset, Set.singleton_subset_iff, SetLike.mem_coe]
+  exact fun _ h ↦ subset_adjoin F _ <| Set.mem_image_of_mem Subtype.val h
```
Comment (alreadydone) [inline Mathlib/FieldTheory/Adjoin.lean:809]: slight golf if you'd like:
```suggestion
  refine ⟨s.image Subtype.val, by simp, SetLike.le_def.mp ?_ hx'⟩
  simp_rw [Finset.coe_image, iSup_le_iff, adjoin_le_iff]
  rintro _ h _ rfl
  exact subset_adjoin F _ ⟨_, h, rfl⟩
```

**Review by alreadydone — APPROVED** [review]

**Comment by jcommelin** [comment]
Please merge @alreadydone's suggestion. After that, and after CI passes, please merge the PR.

bors d+

**Pull request closed without merging**